libraries/ms-common/src/stm32f0xx/interrupt.c

```diff
@@ -0,0 +1,200 @@
+#include "interrupt.h"
+
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "gpio.h"
+#include "gpio_cfg.h"
+#include "status.h"
+#include "stm32f0xx_exti.h"
+#include "stm32f0xx_misc.h"
+#include "stm32f0xx_rcc.h"
+#include "stm32f0xx_syscfg.h"
+
+#define NUM_INTERRUPT_GPIO_CHANNELS 3
```

Comment:
The number of channels is stored as an enum of numbers from 0-31, It just so happens that three are dedicated to GPIO. The only thing I can think of is getting this number to be populated by the preprocessor counting the instances of the substring EXTI within the enum which I don't think is worthwhile... Alternatively, it could be added to the exti or nvic headers.